flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -1082,6 +1110,13 @@ class JobManager(
 
         log.info(s"Using restart strategy $restartStrategy for $jobId.")
 
+        val jobMetrics = jobManagerMetricGroup match {
+          case Some(group) =>
+            group.addJob(jobGraph.getJobID, jobGraph.getName)
```

Comment:
Do you think returning null is a good pattern in the first place? I'd rather return a jobGroup that is neither stored in the jobManagerMetricGroup  nor registers any metrics.